### PR TITLE
split client and server

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: Abdelsalem <abdelsalem.hedhili@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -23,15 +23,32 @@ jobs:
...
         run: mvn --batch-mode -Pjacoco install
+        working-directory: ./case-datasource
 
-      - name: Run SonarCloud analysis
+      - name: Run SonarCloud analysis on datasource client
         run: >
           mvn --batch-mode -DskipTests sonar:sonar
           -Dsonar.host.url=https://sonarcloud.io
           -Dsonar.organization=powsybl-ci-github
           -Dsonar.projectKey=com.powsybl:powsybl-case-server
```
Comment (sBouzols) [inline .github/workflows/maven.yml:35]: will need another projectKey on sonarcloud I think.

File: case-datasource/pom.xml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    <licenses>
+        <license>
+            <name>Mozilla Public License, Version 2.0</name>
+            <url>https://www.mozilla.org/en-US/MPL/2.0/</url>
+        </license>
+    </licenses>
+
+    <scm>
+        <connection>scm:git:https://github.com/powsybl/powsybl-case-server.git</connection>
```
Comment (sBouzols) [inline case-datasource/pom.xml:26]: https://github.com/powsybl/powsybl-case-server.git doesn't exists
https://github.com/powsybl/powsybl-case.git exists... I don't know the severity of this but...

File: case-datasource/pom.xml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        <developerConnection>scm:git:https://github.com/powsybl/powsybl-case-server.git</developerConnection>
```
Comment (sBouzols) [inline case-datasource/pom.xml:27]: same here

File: case-datasource/pom.xml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        <url>https://github.com/powsybl/powsybl-case-server.git</url>
```
Comment (sBouzols) [inline case-datasource/pom.xml:28]: same same

File: .github/workflows/maven.yml (hunk 1 above)
Comment (AbdelHedhili) [reply .github/workflows/maven.yml:35]: agreed

File: case-datasource/pom.xml (hunk 2 above)
Comment (AbdelHedhili) [reply case-datasource/pom.xml:26]: right, updated

File: case-server/pom.xml [hunk 5]
```diff
@@ -118,7 +180,6 @@
         <dependency>
             <groupId>org.springdoc</groupId>
             <artifactId>springdoc-openapi-ui</artifactId>
-            <version>${springdoc-openapi.version}</version>
```
[the remark this answers is not in the document]
Comment (AbdelHedhili) [reply case-server/pom.xml:121]: indeed it is

File: .github/workflows/maven.yml [hunk 6]
```diff
@@ -23,19 +23,27 @@ jobs:
...
+        working-directory: ./case-server
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
 
+      - name: Build case server with Maven
```
Comment (sBouzols) [inline .github/workflows/maven.yml:43]: Is it needed to run the SonarCloud analysis after the step which build the case sserver ?

Comment (AbdelHedhili) [reply .github/workflows/maven.yml:43]: no why ?

**Review by geofjamg — APPROVED** [review]

**Pull request merged**